servers/quarkus-server/src/main/java/org/projectnessie/server/configchecks/ConfigChecks.java

```diff
@@ -0,0 +1,89 @@
+/*
+ * Copyright (C) 2024 Dremio
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.projectnessie.server.configchecks;
+
+import io.quarkus.runtime.Startup;
+import jakarta.enterprise.inject.Produces;
+import jakarta.inject.Inject;
+import jakarta.inject.Singleton;
+import org.projectnessie.quarkus.config.QuarkusServerConfig;
+import org.projectnessie.quarkus.config.VersionStoreConfig;
+import org.projectnessie.quarkus.providers.storage.PersistProvider;
+import org.projectnessie.server.config.QuarkusNessieAuthenticationConfig;
+import org.projectnessie.server.config.QuarkusNessieAuthorizationConfig;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class ConfigChecks {
+  private static final Logger LOGGER = LoggerFactory.getLogger(PersistProvider.class);
+
+  @Inject VersionStoreConfig versionStoreConfig;
+  @Inject QuarkusNessieAuthenticationConfig authenticationConfig;
+  @Inject QuarkusNessieAuthorizationConfig authorizationConfig;
+  @Inject QuarkusServerConfig serverConfig;
+
+  // This function shall be called on startup. The actually returned object is currently
+  // meaningless.
+  @Produces
+  @Singleton
+  @Startup
+  public ConfigCheck configCheck() {
+    if (versionStoreConfig.getVersionStoreType() == VersionStoreConfig.VersionStoreType.IN_MEMORY) {
+      LOGGER.warn(
+          "Non-production IN_MEMORY version-store warning: "
```

Comment:
nit: do we really need this message prefix (similarly the prefixes in the following messages)?